src/components/multipleFileUpload.tsx

```diff
@@ -0,0 +1,371 @@
+    fileUrl: string | StaticImageData;
+    fileName: string;
+  }>;
+
+  if (uploadedFiles) {
+    for (var i = 0; i < uploadedFiles.length; i++) {
+      const obj = {
+        previewType: uploadedFiles[i].fileType,
+        previewUrl: uploadedFiles[i].fileUrl,
+        previewName: uploadedFiles[i].fileName,
+        isDragging: false,
+      };
+
+      defaultPreview.push(obj);
+      defaultFiles.push(uploadedFiles[i]);
+    }
+  }
+
+  const [filesPreview, setFilesPreview] = useState(defaultPreview);
+  const [isUploading, setIsUploading] = useState(false);
+  const [files, setFiles] = useState(defaultFiles);
+
+  let inputRefs = useRef([]);
+  const selectFile = (index: number) => {
+    if (isUploading) {
+      return;
+    }
+    if (inputRefs.current[index]) {
+      (inputRefs.current[index] as HTMLInputElement).click();
+    }
+  };
+
+  const inputsRef = useRef(null);
+  const selectFiles = () => {
+    if (isUploading) {
+      return;
+    }
+    if (inputsRef.current) {
+      (inputsRef.current as HTMLInputElement).click();
+    }
+  };
+
+  const add = (
+    previewType: any,
+    previewUrl: string | StaticImageData | ArrayBuffer | null,
+    previewName: string,
+    isDragging: boolean,
+    file: any
+  ) => {
+    // add file
+    setFiles((files) => {
+      return [...files, file];
+    });
+    // add file preview
+    setFilesPreview((filesPreview) => {
+      return [
+        ...filesPreview,
+        {
+          previewType: previewType,
+          previewUrl: previewUrl,
+          previewName: previewName,
+          isDragging: isDragging,
+        },
+      ];
+    });
+  };
+
+  const update = (
+    previewType: any,
+    previewUrl: string | StaticImageData | ArrayBuffer | null,
+    previewName: string,
+    isDragging: boolean,
+    file: any,
+    index: number
+  ) => {
+    // update file
+    setFiles((files) => {
+      const newFiles = [...files];
+      newFiles[index] = file;
+      return newFiles;
+    });
+    // update file preview
+    setFilesPreview((filesPreview) => {
+      const newFiles = [...filesPreview];
+      newFiles[index].previewType = previewType;
+      newFiles[index].previewUrl = previewUrl;
+      newFiles[index].previewName = previewName;
+      newFiles[index].isDragging = isDragging;
+      return newFiles;
+    });
+  };
+
+  const handleFileChange = (event: any, index: number, action: string) => {
+    const files = event.target.files;
+    for (var i = 0; i < files.length; i++) {
+      try {
+        previewFile(files[i], index + i, action);
+      } catch (error) {
+        console.error("error : ", error);
+      }
+    }
+  };
+
+  const previewFile = (file: any, index: number, action: string) => {
+    var obj = {
+      previewType: "image",
+      previewUrl: "" as string | StaticImageData | ArrayBuffer | null,
+      previewName: file.name,
+      isDragging: false,
+    };
+
+    const reader = new FileReader();
+    reader.onload = () => {
```

Comment:
Let's create reusable code for this as this is used in both components

Reply:
i think it would be good if we keep it as it is cause both is working same but we have managed data into it